modules/core/02-client/keeper/keeper_test.go

```diff
@@ -483,3 +486,114 @@ func (suite *KeeperTestSuite) TestUnsetParams() {
 		suite.chainA.GetSimApp().IBCKeeper.ClientKeeper.GetParams(ctx)
 	})
 }
+
+// TestScheduleIBCClientUpgrade tests that an IBC client upgrade has been properly scheduled
+func (suite *KeeperTestSuite) TestScheduleIBCClientUpgrade() {
+	var (
+		upgradedClientState *ibctm.ClientState
+		oldPlan, plan       upgradetypes.Plan
+		expError            error
+	)
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expPass  bool
+	}{
+		{
+			"valid upgrade proposal",
+			func() {},
+			true,
+		},
+		{
+			"valid upgrade proposal with previous IBC state", func() {
+				oldPlan = upgradetypes.Plan{
+					Name:   "upgrade IBC clients",
+					Height: 100,
+				}
+			}, true,
+		},
+		{
+			"fail: scheduling upgrade with plan height 0",
+			func() {
+				plan.Height = 0
+				expError = sdkerrors.ErrInvalidRequest
```

Comment:
should this be `ibcerrors` now, or no?

Reply:
the reason i'm using `sdkerrors` here is because the error returned is from the sdk `UpgradeKeeper`

Reply:
Makes sense, thanks!